### PR TITLE
[Ahuna] Team sharing section changes in assistant builder

### LINKED ISSUE #3267: Update "Team sharing" in Assistant builder
As per [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22367&mode=design&t=wGVRT1IiZNJ18Mm1-4) and [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22361&mode=design&t=wGVRT1IiZNJ18Mm1-4)
(2 modes depending on role builder or member)

Full [figma](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4050-22251&mode=design&t=Dlhq25QyeMahW2OI-0) (source of truth)

Full [issue](https://github.com/dust-tt/decisions/issues/147#issue-2081773564)

### PR DESCRIPTION
**Pull request by philipperolet**
## Description

Fixes #3267 and #3259

**Note** #3259 renames assistant scope according to new product framing (needed to merge)
## Risks
Normal

## Deploy
Normal

### CODE REVIEW

File: front/components/assistant/AssistantActions.tsx [hunk 1]
```diff
@@ -200,13 +200,13 @@ export function RemoveAssistantFromWorkspaceDialog({
...
           sendNotification({
-            title: `Error removing from Workspace list`,
+            title: `Error removing from Company assistants`,
             description: data.error.message,
             type: "error",
           });
         } else {
           sendNotification({
-            title: `Assistant removed from Workspace list`,
+            title: `Assistant removed from company assistants`,
```
Comment (spolu) [inline front/components/assistant/AssistantActions.tsx:209]: Upper-casing is not consistent?

File: front/components/assistant_builder/TeamSharingSection.tsx [hunk 2]
```diff
@@ -1,280 +1,118 @@
...
+import { useAgentUsage } from "@app/lib/swr";
+
+/*
+ * Note: Non-builders cannot change to/from company assistant
+ */
 export function TeamSharingSection({
   owner,
   agentConfigurationId,
-  initialScope,
   newScope,
```
Comment (spolu) [inline front/components/assistant_builder/TeamSharingSection.tsx:22]: Name newScope is weird without `initialScope` should this be scope?

**Review by spolu — APPROVED** [review]
LGTM 

Left a few comments, feel free to merge once addressed :+1:

File: front/components/assistant_builder/TeamSharingSection.tsx [hunk 3]
```diff
@@ -1,280 +1,118 @@
...
-    useState(false);
-  const router = useRouter();
-
-  if (initialScope !== "private" && newScope === "private") {
-    throw new Error("Cannot change scope back to private");
-  }
-  if (initialScope === "global") {
-    return null;
-  }
+  const agentUsage = agentConfigurationId
```
Comment (fontanierh) [inline front/components/assistant_builder/TeamSharingSection.tsx:30]: I don't think you can do this. You should never conditionally use a hook. Instead the recommended way is to make the swr url be null in cases where you don't want to fetch.

Comment (fontanierh) [reply front/components/assistant_builder/TeamSharingSection.tsx:30]: See here: https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

**Review by fontanierh — CHANGES_REQUESTED** [review]

File: front/components/assistant_builder/TeamSharingSection.tsx (hunk 3 above)
Comment (philipperolet) [reply front/components/assistant_builder/TeamSharingSection.tsx:30]: Ack will change
Weird that I didn't get the Hooks error

Comment (fontanierh) [reply front/components/assistant_builder/TeamSharingSection.tsx:30]: you would get it if the number / order of hooks change. So here, it's fairly unlikely that it gets re-rendered with a non-null `agentConfigurationId` after being rendered with a null `agentConfigurationId`

but it's still good practice to not do this

Comment (philipperolet) [reply front/components/assistant_builder/TeamSharingSection.tsx]: Changed ! Good for you to merge?

**Review by fontanierh — APPROVED** [review]

File: front/components/assistant_builder/TeamSharingSection.tsx (hunk 3 above)
Comment (fontanierh) [reply front/components/assistant_builder/TeamSharingSection.tsx]: Yes all good 🙏

**Pull request merged**